packages/interactivity/src/proxies/context.ts

```diff
@@ -0,0 +1,69 @@
+// TODO: Use the proxy registry to avoid multiple proxies on the same object.
+
+// Store the context proxy and fallback for each object in the context.
+const contextObjectToProxy = new WeakMap();
+const contextProxyToObject = new WeakMap();
+const contextObjectToFallback = new WeakMap();
+
+const descriptor = Reflect.getOwnPropertyDescriptor;
+
+const contextHandlers: ProxyHandler< object > = {
+	get: ( target: object, k: string ) => {
+		const fallback = contextObjectToFallback.get( target );
+		// Always subscribe to prop changes in the current context.
+		const currentProp = target[ k ];
+
+		// Return the inherited prop when missing in target.
+		if ( ! ( k in target ) && k in fallback ) {
+			return fallback[ k ];
+		}
+
+		/*
+		 * For other cases, return the value from target, also
+		 * subscribing to changes in the parent context when the current
+		 * prop is not defined.
+		 */
+		return k in target ? currentProp : fallback[ k ];
+	},
+	set: ( target, k, value ) => {
+		const fallback = contextObjectToFallback.get( target );
+		const obj = k in target || ! ( k in fallback ) ? target : fallback;
+		obj[ k ] = value;
```

Comment:
`Reflect.set` here as well?

Reply:
Same answer as in https://github.com/WordPress/gutenberg/pull/64713#discussion_r1728853160. 😄